langwatch/src/server/background/workers/evaluationsWorker.ts

```diff
@@ -77,6 +81,133 @@ export async function runEvaluationJob(
   });
 }
 
+/**
+ * Check if any mapping has type "thread"
+ * Single Responsibility: Detect if thread-based mappings are present
+ */
+const hasThreadMappings = (mappingState: MappingState): boolean => {
+  return Object.values(mappingState.mapping).some(
+    (mapping) => "type" in mapping && mapping.type === "thread"
+  );
+};
+
+/**
+ * Extract selected fields from traces based on thread mapping configuration
+ * Single Responsibility: Transform traces array into field values based on selectedFields
+ */
+const extractThreadFields = (
```

Comment:
shouldn't this be called `extractTracesFields` then?

and `selectedFields` perhaps should be of type `(keyof typeof TRACE_MAPPINGS)[]`